### ISSUE TITLE
Startup on Windows doesn't draw decorators

### ISSUE THREAD

**Issue by superlou** [lapce/lapce#1014]
When running the latest nightly (though it's been this way for about a month), Lapce doesn't draw the Window border/decorators correctly on startup.

![image](https://user-images.githubusercontent.com/709695/187439490-11abfacc-254f-461a-9869-d4397bc7a50a.png)

Resizing the window in any way fixes the border.

![image](https://user-images.githubusercontent.com/709695/187439897-5f8903d2-a37f-4a92-aee8-f7fa80214047.png)

**Comment by superlou** [comment]
Sorry, I might have used the wrong term. I believe it is the titlebar (see above pictures). The region with the "open folder" drop-down and Window min/max/close isn't visible.

**Comment by panekj** [comment]
Could you give output of `$PSVersionTable` from `powershell`?

**Comment by superlou** [comment]
```
PS C:\> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.19041.1682
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.1682
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by superlou** [comment]
Do you mean that you don't have a Windows machine to test on, or it is not occurring on another Windows machine? I'll see what I can test out, though I do all my development on Linux at home. Was just filing the issue in case someone else has also run into it.

**Comment by panekj** [comment]
> Do you mean that you don't have a Windows machine to test on, or it is not occurring on another Windows machine?

![image](https://user-images.githubusercontent.com/50457605/187455949-6abbfe96-aa3b-4b4b-86b0-8e88d59bef6c.png)

**Comment by superlou** [comment]
Thanks. Looks like it's not reproduced on another Windows 10(?) computer. I'll try and see if I can find some other candidates to test on.

**Comment by panekj** [comment]
> Windows 10(?) computer.

Whole titlebar thing was done solely on my Windows 10. I don't have anything else (other than Win11 VM for testing)

**Comment by superlou** [comment]
Some additional system specs:

OS: Microsoft Windows 10 Pro
Version: 10.0.19044 Build 19044
System Model: Dell OptiPlex 7070
Processor: Intel(R) Core(TM) i5-9500 CPU @ 3.00GHz, 3000 Mhz, 6 Core(s), 6 Logical Processor(s)
Graphics: Intel(R) UHD Graphics 630
Graphics Driver Version: 30.0.100.9864

It's definitely not always a problem on Windows. There must be something about my system exacerbating it.

**Comment by seomwan** [comment]
I don't have this exact bug, but rather the title bar is offset from the top:

![image](https://user-images.githubusercontent.com/68874610/188365316-1e54a122-3237-4c61-9b41-108d8e6d0560.png)
![image](https://user-images.githubusercontent.com/68874610/188365340-cca56238-ed8c-49d7-a3ec-13e59971d559.png)

Hopefully you can see the hovering is offset too.

I further noticed that it may also be offset from the left?
![image](https://user-images.githubusercontent.com/68874610/188365575-f9654b6b-0361-42ec-bc43-c637a5dddd8e.png)

Here is my system info

```
PS C:\> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.19041.1682
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.1682
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by seomwan** [comment]
Should this be a separate bug report?

Here's the log with `RUST_LOG = debug`, for a second only so as not to clutter with useless stuff.

https://pastebin.com/xAYwTTym

Nothing seems unusual, mostly lots of:
```
[INFO] lifecycle;
[INFO] paint;
[INFO] event;
[INFO] update;
[INFO] layout;
```

**Comment by superlou** [comment]
@seomwan, does everything return to normal behavior after resizing the window?

**Comment by seomwan** [comment]
FYI, I was using release 0.2.0 [Lapce-windows.msi](https://github.com/lapce/lapce/releases/download/v0.2.0/Lapce-windows.msi)

Just realized I didn't mention that anywhere.

**Comment by superlou** [comment]
@seomwan, If resizing/redrawing lapce doesn't resolve the display issue,  I suspect your issue is different.

**Comment by dflock** [comment]
This also happens to me:

```console
> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.2.5
PSEdition                      Core
GitCommitId                    7.2.5
OS                             Microsoft Windows 10.0.19044
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

> Get-Winver

Name                           Value
----                           -----
DisplayVersion                 21H2
ProductName                    Windows 10 Enterprise
CurrentBuildNumber             19044
KeyName                        HKLM:\SOFTWARE\WOW6432Node\Microsoft\Windows NT\CurrentVersion
Version                        10.0.19044.0
VersionString                  Microsoft Windows NT 10.0.19044.0
```

Currently using a Lenovo Thinkpad X1, with Intel Xe integrated Graphics:

![image](https://user-images.githubusercontent.com/47756/188979707-28d0b9e9-2288-46eb-94f1-8c1b0762636d.png)

**Comment by CGMossa** [comment]
I too am having these "offsets" issues.

**Comment by panekj** [comment]
For anyone who has "black border", your issue is https://github.com/lapce/lapce/issues/1083

This issue (https://github.com/lapce/lapce/issues/1014) is about missing titlebar

**Comment by superlou** [comment]
@dflock, good call. I also have Intel drivers:

![image](https://user-images.githubusercontent.com/709695/189660108-66c4bd3b-5990-4f50-8067-7afc81579116.png)

**Comment by dflock** [comment]
Can confirm that switching off custom title bars fixes this for me.

**Comment by sysmat** [comment]
for me nothings work,
- none of buttons work(open folder, ...), not even close

 ```
$PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.2.6
PSEdition                      Core
GitCommitId                    7.2.6
OS                             Microsoft Windows 10.0.19044
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by bugadani** [comment]
I don't what changed but the latest nightly works for me as expected.

**Comment by superlou** [comment]
I still have the issue with the latest nightly.

**Comment by bugadani** [comment]
No driver updates from what I can see. Maybe I just got lucky :(

**Comment by Hummel37** [comment]
I have the same problem but with Windows 11 😕

**Comment by superlou** [comment]
@Hummel37, what graphics card and drivers do you have?

**Comment by Hummel37** [comment]
![image](https://user-images.githubusercontent.com/54278236/195273167-b00d605c-6db9-405c-b114-46fc88edcd4b.png)

**Comment by Hyperchaotic** [comment]
I also see it with an Intel i7-9750H/UHD630. If switch Lapce to the discrete Nvidia 2060mq card it doesn't happen. 

Running latest source built from git right now.

    Driver Details Up to date
    Provider Intel Corporation
    Version 31.0.101.2111
    Date 7/19/2022

    Device Details
    Adapter Compatibility Intel Corporation
    Video Processor Intel® UHD Graphics Family

Name                           Value
----                           -----
PSVersion                      5.1.22000.832
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.22000.832
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1

**Comment by glymehrvrd** [comment]
Any progress here? meet the same problem, and can be fixed by turning off custom titlebar or resize the window, quite annoying.

**Comment by panekj** [comment]
> Any progress here? Meet the same problem, and can be fixed by turning off custom titlebar or resize the window, quite annoying.

no, if there will be any progress there will be comment about it

**Comment by almondtools** [comment]
Maybe related:

![grafik](https://github.com/lapce/lapce/assets/2934333/7a0a7aa8-4382-4910-9c8a-1fa3a83f2531)

The title bar appears with padding to the top. Clicking on the title bar elements has no effect, clicking the area above the title bar works. The click-handlers appear to be located at the correct location, but the visual elements of the title bar not.

**Comment by panekj** [comment]
> Maybe related:
> 
> The title bar appears with padding to the top. Clicking on the title bar elements has no effect, clicking the area above the title bar works. The click-handlers appear to be located at the correct location, but the visual elements of the title bar not.

Not related, that's #1083
